src/bidiTab/bidiTab.ts

```diff
@@ -42,49 +47,31 @@ declare global {
 
     // Set from the server side if verbose logging is required.
     sendDebugMessage?: ((message: string) => void) | null;
-
-    // `window.setSelfTargetId` is called via `Runtime.evaluate` from the server side.
-    setSelfTargetId: (targetId: string) => void;
   }
 }
 
-// Initiate `setSelfTargetId` as soon as possible to prevent race condition.
-const waitSelfTargetIdPromise = waitSelfTargetId();
-
-void (async () => {
-  generatePage();
+generatePage();
+const mapperTabToServerTransport = new WindowBidiTransport();
+const cdpTransport = new WindowCdpTransport();
 
-  // Needed to filter out info related to BiDi target.
-  const selfTargetId = await waitSelfTargetIdPromise;
+/**
+ * Set `window.runMapper` to a function which launches the BiDi mapper instance.
+ * @param selfTargetId Needed to filter out info related to BiDi target.
+ */
+window.runMapperInstance = async (selfTargetId) => {
+  console.log('Launching Mapper instance with selfTargetId: ', selfTargetId);
```

Comment:
nit: `console.log` already adds a space, so currently this adds two spaces

Reply:
will remove in the next iteration

Reply:
oh, the branch was outdated anyways. Removed the space.